src/sqlfluff/rules/L043.py

```diff
@@ -161,29 +193,61 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 if not column_reference_segment:
                     return None
 
-                # Check if we can reduce the CASE expression to a single coalesce function.
-                if (
-                    not is_not_prefix
-                    and column_reference_segment.raw_upper == else_expression.raw_upper
-                ):
-                    coalesce_arg_1 = else_expression
-                    coalesce_arg_2 = then_expression
+                if else_clauses:
+                    else_expression = else_clauses.children(sp.is_type("expression"))[0]
+                    # Check if we can reduce the CASE expression to a single coalesce function.
+                    if (
+                        not is_not_prefix
+                        and column_reference_segment.raw_upper
+                        == else_expression.raw_upper
+                    ):
+                        coalesce_arg_1 = else_expression
+                        coalesce_arg_2 = then_expression
+                    elif (
+                        is_not_prefix
+                        and column_reference_segment.raw_upper
+                        == then_expression.raw_upper
+                    ):
+                        coalesce_arg_1 = then_expression
+                        coalesce_arg_2 = else_expression
+                    else:
+                        return None
+
+                    if coalesce_arg_2.raw_upper == "NULL":
+                        # Can just specify the column on it's own
+                        # rather than using a COALESCE function.
+                        return LintResult(
+                            anchor=condition_expression,
+                            fixes=self._column_only_fix_list(
+                                context,
+                                column_reference_segment,
+                            ),
+                            description=f"Unnecessary CASE statement. Just use column '{column_reference_segment.raw}'.",
+                        )
+
+                    return LintResult(
+                        anchor=condition_expression,
+                        fixes=self._coalesce_fix_list(
+                            context,
+                            coalesce_arg_1,
+                            coalesce_arg_2,
+                        ),
+                        description="Unnecessary CASE statement. Use COALESCE function instead.",
+                    )
                 elif (
-                    is_not_prefix
-                    and column_reference_segment.raw_upper == then_expression.raw_upper
+                    column_reference_segment.raw_segments_upper
+                    == then_expression.raw_segments_upper
```

Comment:
N.B. this clause also catches the case where there is no ELSE clause, since that is equivalent to ELSE NULL.
Reason for including is that although this statement is fixed by L043 in isolation:
```sql
select
    foo,
    case
        when
            bar is not null then bar
        else null
    end as test
from baz;
```
In practice the `else null` is first removed by L035 leaving us with:
```sql
select
    foo,
    case
        when
            bar is not null then bar
    end as test
from baz;
```
And this addition makes L043 smart enough to finish the cleanup regardless.